base/linalg/bidiag.jl

```diff
@@ -272,13 +272,13 @@ end
 
 function diag(M::Bidiagonal{T}, n::Integer=0) where T
     if n == 0
-        return M.dv
-    elseif n == 1
-        return M.uplo == 'U' ? M.ev : zeros(T, size(M,1)-1)
-    elseif n == -1
-        return M.uplo == 'L' ? M.ev : zeros(T, size(M,1)-1)
+        return copy(M.dv)
```

Comment:
Should change this to `copy!(similar(M.dv, length(M.dv)), M.dv)` to make certain we return the same type, since we then call the same `similar` method no matter what `n` is.